### PR TITLE
Faulhaber support

### PR DESCRIPTION
**Pull request by pi-r-p**

### CODE REVIEW

File: srcs/pressure_controller.cpp [hunk 1]
```diff
@@ -657,17 +727,36 @@ PressureController::pidPatient(int32_t targetPressure, int32_t currentPressure,
...
+    if (error > -10) {
+        // change of state
+        if (patientPIDFastMode) {
+            patientIntegral = -1000;
+        }
+        patientPIDFastMode = false;
+    }
+
+    // Dont compute derivative
+    int32_t derivative = 0;
```
Comment (dsferruzza) [inline srcs/pressure_controller.cpp:739]: Why keeping `PID_PATIENT_KD` if we don't compute derivative at all? Why not continue computing derivative and have `PID_PATIENT_KD = 0` in faulhaber?

File: srcs/pressure_controller.cpp [hunk 2]
```diff
@@ -629,25 +703,62 @@ int32_t PressureController::pidBlower(int32_t targetPressure, int32_t currentPre
...
-    uint32_t blowerAperture =
+    if (blowerPIDFastMode) {
+        blowerAperture = minAperture;
+    } else {
+        int32_t coefficientI = PID_BLOWER_KI;
+        /*if (abs(error) < 10) {
+            coefficientI = PID_BLOWER_KI * 2;
+        } else {
+            coefficientI = PID_BLOWER_KI / 2;
+        }*/
```
Comment (jabby) [inline srcs/pressure_controller.cpp:727]: Can we delete these lines?

File: srcs/pressure_controller.cpp [hunk 3]
```diff
@@ -629,25 +703,62 @@ int32_t PressureController::pidBlower(int32_t targetPressure, int32_t currentPre
...
+    Serial.print(currentPressure);
+    Serial.print(",");
+    Serial.print((PID_BLOWER_KP * error) / 1000);
+    Serial.print(",");
+    Serial.print(blowerIntegral);
+    Serial.print(",");
+    Serial.print(PID_BLOWER_KD*derivative/1000);
+    Serial.print(",");
+    Serial.print(blowerAperture);
+    Serial.println();
```
Comment (jabby) [inline srcs/pressure_controller.cpp:760]: Should we kept this? If so can we use `DBG_DO` around each line?

File: includes/parameters.h [hunk 4]
```diff
@@ -93,23 +93,23 @@ static const int32_t PID_PATIENT_SAFETY_PEEP_OFFSET = 0;
...
-static const int32_t PID_BLOWER_INTEGRAL_MIN = -1000;
-
-static const int32_t PID_PATIENT_KP = 4;
-static const int32_t PID_PATIENT_KI = 20;
-static const int32_t PID_PATIENT_KD = 10;
+static const int32_t PID_BLOWER_KP = 2400;//2400
+static const int32_t PID_BLOWER_KI = 34;  //34
+static const int32_t PID_BLOWER_KD = 50;	  //42
+static const int32_t PID_BLOWER_INTEGRAL_MAX = 10000;//600
+static const int32_t PID_BLOWER_INTEGRAL_MIN = -10000;//-600
```
Comment (dsferruzza) [inline includes/parameters.h:100]: These comments are useless, right?

File: srcs/pressure_controller.cpp [hunk 5]
```diff
@@ -146,9 +159,16 @@ void PressureController::initRespiratoryCycle() {
 
     // Reset PID integrals
     blowerIntegral = 0;
-    blowerLastError = INVALID_ERROR_MARKER;
+    #if VALVE_TYPE == VT_FAULHABER
+      blowerLastError =  m_maxPlateauPressureCommand - m_minPeepCommand ;
+    #else
+      blowerLastError =  m_maxPeakPressureCommand - m_minPeepCommand ;
+    #endif
```
Comment (dsferruzza) [inline srcs/pressure_controller.cpp:166]: Could you add a comment to explain why?

File: srcs/pressure_controller.cpp [hunk 6]
```diff
@@ -394,11 +433,17 @@ void PressureController::updatePhase(uint16_t p_centiSec) {
...
 
+        #if VALVE_TYPE == VT_FAULHABER
+                int32_t pressureToTest =  m_maxPlateauPressureCommand;
+        #else
+                int32_t pressureToTest =  m_maxPeakPressureCommand;
+        #endif
+
         if ((p_centiSec < ((m_centiSecPerInhalation * 80u) / 100u))
-            && (m_pressure < (m_maxPeakPressureCommand - 10u))) {
+            && (m_pressure < (pressureToTest))) {
```
Comment (dsferruzza) [inline srcs/pressure_controller.cpp:443]: Useless brackets here

File: srcs/pressure_controller.cpp [hunk 7]
```diff
@@ -627,27 +718,67 @@ void PressureController::setSubPhase(CycleSubPhases p_subPhase, uint16_t p_centi
 
 int32_t PressureController::pidBlower(int32_t targetPressure, int32_t currentPressure, int32_t dt) {
     // Compute error
-    int32_t error = targetPressure - currentPressure;
-
-    // Compute integral
-    blowerIntegral = blowerIntegral + ((PID_BLOWER_KI * error * dt) / 1000000);
-    blowerIntegral = max(PID_BLOWER_INTEGRAL_MIN, min(PID_BLOWER_INTEGRAL_MAX, blowerIntegral));
+    //int32_t error = targetPressure + m_plateauPressureOffset - currentPressure;
```
Comment (dsferruzza) [inline srcs/pressure_controller.cpp:721]: Useless comment?

File: srcs/pressure_controller.cpp [hunk 8]
```diff
@@ -627,27 +718,67 @@ void PressureController::setSubPhase(CycleSubPhases p_subPhase, uint16_t p_centi
...
+    m_lastBlowerPIDErrorIndex++;
+    if (m_lastBlowerPIDErrorIndex >= NUMBER_OF_SAMPLE_BLOWER_DERIVATIVE_MOVING_MEAN){
+      m_lastBlowerPIDErrorIndex = 0;
+    }
+    
+    for (uint8_t i = 0u; i < NUMBER_OF_SAMPLE_BLOWER_DERIVATIVE_MOVING_MEAN; i++) {
+        totalValues += m_lastBlowerPIDError[i];
+    }
+    smoothError = totalValues / NUMBER_OF_SAMPLE_BLOWER_DERIVATIVE_MOVING_MEAN;
+    //Serial.println(smoothError);
```
Comment (dsferruzza) [inline srcs/pressure_controller.cpp:758]: Useless comment?

File: srcs/pressure_controller.cpp [hunk 9]
```diff
@@ -627,27 +718,67 @@ void PressureController::setSubPhase(CycleSubPhases p_subPhase, uint16_t p_centi
...
+
+    derivative = ((dt == 0))
+                         ? 0
+                         : ((1000000 * (blowerLastError - smoothError)) / dt);
+
+    blowerIntegral = blowerIntegral + ((coefficientI * error * dt) / 1000000);
+    blowerIntegral =
+        max(PID_BLOWER_INTEGRAL_MIN, min(PID_BLOWER_INTEGRAL_MAX, blowerIntegral));
+    int32_t blowerCommand = (PID_BLOWER_KP * error) / 1000 + blowerIntegral + PID_BLOWER_KD*derivative/1000;// + PID_BLOWER_KA*acceleration;  // Command computation
```
Comment (dsferruzza) [inline srcs/pressure_controller.cpp:767]: Useless comment?

**Comment by pi-r-p** [comment]
All fixed by @Benhalor and marked resolved.

**Pull request merged**